clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -271,11 +268,13 @@
  *     consumer.assign(Arrays.asList(partition0, partition1));
  * </pre>
  *
- * The group that the consumer specifies is still used for committing offsets, but now the set of partitions will only
- * be changed if the consumer specifies new partitions, and no attempt at failure detection will be made.
+ * Once assigned, you can call {@link #poll(long) poll} in a loop, just as in the preceding examples to consume
+ * records. The group that the consumer specifies is still used for committing offsets, but now the set of partitions
+ * will only be changed if the consumer specifies new partitions in another call to {@link #assign(Collection) assign},
+ * and no attempt at failure detection will be made.
```

Comment:
I think the phrase "failure detection" here is a little vague. I think what we're trying to get at is that the coordinator won't reassign the partitions. How about this instead:

> Since manual partition assignment does not use group coordination, consumer failures will not cause assigned partitions to be rebalanced. Each consumer acts independently even if it shares a groupId with another consumer. To avoid offset commit conflicts, you should usually ensure that the groupId is unique for each consumer instance.